pwiz_tools/Skyline/TestPerf/AlphapeptdeepBuildLibraryTest.cs

```diff
@@ -0,0 +1,430 @@
+
+                        Assert.AreEqual(string.Format(ToolsUIResources.PythonInstaller_Requesting_Administrator_elevation),
+                            longPathDlg.Message);
+
+                        if (PythonInstaller.IsRunningElevated())
+                        {
+                            confirmDlg = AbstractFunctionalTest.ShowDialog<MessageDlg>(pythonDlg.OkDialog, WAIT_TIME);
+                            ConfirmPythonSuccess(confirmDlg);
+
+                        }
+                        else
+                        {
+                            Assert.Fail($@"Error: Cannot finish {_toolName}BuildLibraryTest because {PythonInstaller.REG_FILESYSTEM_KEY}\{PythonInstaller.REG_LONGPATHS_ENABLED} is not set and have insufficient permissions to set it");
+                        }
+                    }
+                    else
+                    {
+                        Console.WriteLine(@"Info: LongPathsEnabled registry key is already set to 1");
+                        AbstractFunctionalTest.OkDialog(pythonDlg, pythonDlg.OkDialog);
+                        confirmDlg = AbstractFunctionalTest.WaitForOpenForm<MessageDlg>();
+                        ConfirmPythonSuccess(confirmDlg);
+
+                    }
+
+
+                }, dlg => {
+                    dlg.Close();
+                });
+                if (_undoRegistry)
+                {
+                    PythonInstaller.DisableWindowsLongPaths();
+                }
+
+                return true;
+            }
+
+            return false;
+        }
+        /// <summary>
+        /// Checks for Python prerequisite
+        /// </summary>
+        /// <param name="buildLibraryDlg">Build Library dialog</param>
+        /// <returns></returns>
+        public bool HavePythonPrerequisite(BuildLibraryDlg buildLibraryDlg)
+        {
+            bool havePythonPrerequisite = false;
+            AbstractFunctionalTest.RunUI(() => { havePythonPrerequisite = buildLibraryDlg.PythonRequirementMet(); });
+            return havePythonPrerequisite;
+        }
+
+        public bool HaveNvidiaSoftware()
+        {
+            return PythonInstaller.NvidiaLibrariesInstalled();
+        }
+
+        public bool HaveNvidiaHardware()
+        {
+            return PythonInstaller.TestForNvidiaGPU() == true;
+        }
+        /// <summary>
+        /// Runs Nvidia Dialog
+        /// </summary>
+        /// <param name="nvidiaDlg">Nvidia Detected Dialog</param>
+        /// <param name="pythonDlg">Python Installer Dialog</param>
+        /// <param name="clickNo">true clicks No, false clicks Yes, null clicks Cancel to Nvidia Detected Dialog</param>
+        private void RunNvidiaDialog(MultiButtonMsgDlg nvidiaDlg, MultiButtonMsgDlg pythonDlg, bool? clickNo = true)
+        {
+            if (clickNo == true)
+            {
+
+                // Say 'No'
+                AbstractFunctionalTest.RunLongDlg<AlertDlg>(nvidiaDlg.ClickNo, confirmDlg =>
```

Comment:
You only need to use "RunLongDlg" if you are bringing up a dialog which in turn brings up more dialogs.
Usually, you would just do:
`RunDlg<AlertDlg>(nvidiaDlg.ClickNo, confirmDlg=>ConfirmPythonSuccess(confirmDlg));`
However, that only works if eveything that "ConfirmPythonSuccess" does can be run on the event thread.

ConfirmPythonSuccess is doing the following:
`OkDialog(confirmDlg, confirmDlg.OkDialog);`
which cannot be run on the event thread.
That line in "ConfirmPython" should be changed to:
confirmDlg.OkDialog();
and all of the callers of ConfirmPython should be changed so that they are being called from the event thread.

I can make that code change if it's not clear what I am talking about.